lms/templates/partials/analytics.html

```diff
@@ -46,25 +17,73 @@
   else:
     courseName = 'Not in course'
   endif
-
-  appsemblerLmsTrackingCode = get_global_settings().get('appsembler_lms_tracking_code')
 %>
 
-<script>
-  % if get_global_settings().get('appsembler_lms_tracking_code'):
-    ga('create', '${appsemblerLmsTrackingCode}', 'auto', 'appsemblerLmsTracker');
-    ga('appsemblerLmsTracker.send', 'pageview', {
-      %if USER_EMAIL_HASH:
-        'userId': '${USER_EMAIL_HASH}',
-      %endif
-      'dimension1': '${siteOrganization}',
-      'dimension2': '${userType}',
-      'dimension3': '${courseName}',
-      'dimension4': 'LMS'
-    });
-  % endif
-</script>
-<!-- /new analytics -->
+% if (get_global_settings().get('customer_gtm_enabled') and (get_global_settings().get('customer_gtm_id') != "")):
+  <%
+    customerGtmId = get_global_settings().get('customer_gtm_id')
+  %>
+  <script>
+    dataLayer = [{
+      'visitorType': '${userType}',
+      'courseName': '${courseName}'
+    }];
+  </script>
+  <!-- Google Tag Manager -->
+  <script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+  new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+  j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+  'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+  })(window,document,'script','dataLayer','${customerGtmId}');</script>
```

Comment:
In order to hack this, so long as we're not doing input checking on the AMC side, I could enter something for my GTM ID like: 
```
000'); (function();do_evil_things('bwahaha'));('
```

and get away with doing evil things.

Reply:
Great catch Aaron! Thank you.

Open edX has a pretty good guideline for Mako templates security. I think we should enable it via CI sometime soon.

For now here's what we need to change:

```suggestion
  })(window,document,'script','dataLayer','${customerGtmId | n, js_escaped_string}');</script>
```

We also need to add an import::

```
<%! from openedx.core.djangolib.js_utils import js_escaped_string %>
```